Tests/AutoCollection/Statsbeat.tests.ts

```diff
@@ -0,0 +1,209 @@
+        });
+    });
+
+    describe("#Resource provider property", () => {
+        it("unknown resource provider", (done) => {
+            const statsBeat: Statsbeat = new Statsbeat(config);
+            statsBeat["_getResourceProvider"](() => {
+                assert.equal(statsBeat["_resourceProvider"], "unknown");
+                done();
+            });
+
+        });
+
+        it("app service", (done) => {
+            const statsBeat: Statsbeat = new Statsbeat(config);
+            var newEnv = <{ [id: string]: string }>{};
+            newEnv["WEBSITE_SITE_NAME"] = "Test Website";
+            var originalEnv = process.env;
+            process.env = newEnv;
+            statsBeat["_getResourceProvider"](() => {
+                process.env = originalEnv;
+                assert.equal(statsBeat["_resourceProvider"], "appsvc");
+                done();
+            });
+
+        });
+
+        it("Azure Function", (done) => {
+            const statsBeat: Statsbeat = new Statsbeat(config);
+            var newEnv = <{ [id: string]: string }>{};
+            newEnv["FUNCTIONS_WORKER_RUNTIME"] = "nodejs";
+            var originalEnv = process.env;
+            process.env = newEnv;
+            statsBeat["_getResourceProvider"](() => {
+                process.env = originalEnv;
+                assert.equal(statsBeat["_resourceProvider"], "function");
+                done();
+            });
+        });
+
+        it("Azure VM", (done) => {
+            const statsBeat: Statsbeat = new Statsbeat(config);
+            var newEnv = <{ [id: string]: string }>{};
+            var originalEnv = process.env;
+            process.env = newEnv;
+            let interceptor = nock("http://169.254.169.254")
+                .get("/metadata/instance/compute", (body: string) => {
+                    return true;
+                });
+            interceptor.reply(200, {
+                "vmId": "testId",
+                "subscriptionId": "testsubscriptionId",
+                "osType": "testOsType"
+            });
+            statsBeat["_getResourceProvider"](() => {
+                process.env = originalEnv;
+                assert.equal(statsBeat["_resourceProvider"], "vm");
+                done();
+            });
+        });
+    });
+
+    describe("#trackStatsbeatMetrics", () => {
+        it("It adds correct properties to custom metric", (done) => {
+            const statsBeat: Statsbeat = new Statsbeat(config);
+            statsBeat.enable(true);
+            const spy = sandbox.spy(statsBeat["_sender"], "send");
+            statsBeat.countRequest(123, true);
+            statsBeat.trackStatsbeatMetrics().then(() => {
+                assert.equal(spy.callCount, 1, "should call sender");
+                let envelope = spy.args[0][0][0];
+                assert.equal(envelope.name, "Statsbeat");
+                assert.equal(envelope.iKey, "c4a29126-a7cb-47e5-b348-11414998b11e");
```

Comment:
it will be better if you don't test against the prod statsbeat ikey.  make it configurable will help.

Reply:
and it will minimize testing data infiltering with the customer's stats data.